arches/app/models/graph.py

```diff
@@ -2154,6 +2154,20 @@ def validate(self):
                         999,
                     )
 
+        # validates that a graph slug has not changed on a published graph
+        if (
+            self.publication_id
+            and not self.source_identifier_id
+            and self.get_published_graph().serialized_graph is not None
+            and self.slug != self.get_published_graph().serialized_graph["slug"]
+        ):
+            raise GraphValidationError(
+                _(
+                    "You cannot change the slug of a published graph. Please create a new publication to edit graph slug."
+                ),
+                1018,
```

Comment:
I wonder if it's ok to change the graph slug even between publications? Even that seems dubious. After all, different publications are meant to ostensibly manage the same intent of the model (which the slug is just shorthand for).

Reply:
It seems like someone may want to change the slug between publications if, for example, they just have a typo or prefer a better name

Reply:
yeah it seems like we may not want to enforce this? Making this work moot? I can't imagine a slug change affecting the graph shape.

Reply:
yeah, that certainly makes sense to me.

Reply:
That makes sense, but it's contradictory to 

> It seems like someone may want to change the slug between publications if, for example, they just have a typo or prefer a better name

no?

Reply:
ah -- my misunderstand of ` between publications` . this all lgtm!

Reply:
Was about to respond to the issue comment but I see it's been sorted out here!

Reply:
I think documentation explaining the importance of the "graph slug" and how it impacts api usage etc... is going to be very important.  Changing graph slugs, even to fix typos, can have detrimental impacts to other apps using those apis and other integrations.